distro/commands/docs.go

```diff
@@ -0,0 +1,53 @@
+package commands
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"os"
+)
+
+type filesystem interface {
+	Mkdir(name string, perm os.FileMode) error
+	Stat(name string) (os.FileInfo, error)
+}
+
+type localFs struct{}
+
+func (localFs) Mkdir(name string, perm os.FileMode) error {
+	return os.Mkdir(name, perm)
+}
+func (localFs) Stat(name string) (os.FileInfo, error) {
+	return os.Stat(name)
+}
+
+func DistroDocs(rootCommand *cobra.Command, fs filesystem) *cobra.Command {
+	var directory string
+
+	command := &cobra.Command{
+		Use:    "docs",
+		Short:  "generate riff-distro command documentation",
+		Hidden: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			fileinfo, err := fs.Stat(directory)
+			if os.IsNotExist(err) {
+				if err = fs.Mkdir(directory, 0744); err != nil {
+					return err
+				}
+				fileinfo, err = fs.Stat(directory)
```

Comment:
Without the line above, the doc generation will fail if the docs folder is not present, which makes sense.
I do not understand how the other impl does not need that.